nrf-hal-common/src/saadc.rs

```diff
@@ -74,23 +80,129 @@ impl Saadc {
             .write(|w| w.oversample().variant(oversample));
         saadc.samplerate.write(|w| w.mode().task());
 
-        saadc.ch[0].config.write(|w| {
+        // Calibrate
+        saadc.tasks_calibrateoffset.write(|w| unsafe { w.bits(1) });
+        while saadc.events_calibratedone.read().bits() == 0 {}
+
+        Saadc { saadc }
+    }
+
+    pub fn channel(&mut self, n: usize) -> Channel {
+        Channel {
+            saadc: &mut self.saadc,
+            channel: n,
+        }
+    }
+}
+
+pub struct Channel<'a> {
+    saadc: &'a mut SAADC,
+    channel: usize,
+}
+
+impl<'a> Channel<'a> {
+    pub fn configure(
```

Comment:
Does this *have* to be called for every channel that is used? What's the default state that would be used by the existing impls if this isn't called?